clang/include/clang/StaticAnalyzer/Core/PathSensitive/SMTConstraintManager.h

```diff
@@ -278,6 +278,13 @@ class SMTConstraintManager : public clang::ento::SimpleConstraintManager {
     if (const SymbolCast *SC = dyn_cast<SymbolCast>(Sym))
       return canReasonAbout(SVB.makeSymbolVal(SC->getOperand()));
 
+    // If a UnarySymExpr is encountered, the Z3
+    // wrapper does not support those. So indicate Z3 does not
+    // support those and return.
+    if (dyn_cast<UnarySymExpr>(Sym)) {
```

Comment:
You can use "isa" here instead of dyn_cast since we're not really using the casted value anyway.